src/main/scala/com/microsoft/hyperspace/package.scala

```diff
@@ -47,6 +48,8 @@ package object hyperspace {
       disableHyperspace
       sparkSession.sessionState.experimentalMethods.extraOptimizations ++=
         hyperspaceOptimizationRuleBatch
+      sparkSession.sessionState.experimentalMethods.extraStrategies ++=
+        BucketUnionStrategy :: Nil
```

Comment:
nit: When `disableHyperspace` is called, should we also remove the extraStrategies we injected?

Reply:
Same for `isHyperspaceEnabled` on whether we should double-check that the strategy has been inserted correctly.

Reply:
Thanks! I think we should remove the extraStrategies in disbleHyperspace() to avoid duplicates by enabling/disabling hyperspace.